includes/admin/class-sensei-learners-admin-bulk-actions-view.php

```diff
@@ -272,6 +272,16 @@ private function get_learner_html( $learner ) {
 	private function get_learners( $args ) {
 		$query             = new Sensei_Db_Query_Learners( $args );
 		$learners          = $query->get_all();
+
+		if ( isset( $_GET['orderby'] ) && 'last_activity_date' === $_GET['orderby'] ) {
```

Comment:
Hey @mikeyarce 👋  
Here, I think, we sort students only on the current page, because `$query->get_all()` returns only part of the list (limited with `per_page` and `offset`).
That will lead to surprising results :) Have you tested it when you have two or three pages of students?

Reply:
Thanks @merkushin !  
What do you think about alerting the SQL query [here](https://github.com/Automattic/sensei/blob/trunk/includes/class-sensei-db-query-learners.php#L75-L83) when the `orderby` is set to `last_activity_date`?  

Feels a bit messy but something like this?

```
$sql = "
	SELECT SQL_CALC_FOUND_ROWS
		`u`.`ID` AS 'user_id',
		`u`.`user_nicename`,
		`u`.`user_login`,
		`u`.`user_email`,
		'' AS 'course_statuses',
		0 AS 'course_count',";

if ( 'last_activity_date' === $this->order_by ) {
	$sql .= "
	COUNT(`c`.`comment_approved`) AS 'course_count', (
		SELECT MAX(cm.comment_date_gmt)
		FROM {$wpdb->comments} cm
		WHERE cm.user_id = u.ID
		AND cm.comment_approved IN ('complete', 'passed', 'graded')
		AND cm.comment_type = 'sensei_lesson_status'
	) AS last_activity_date";
}

$sql .= " FROM `{$wpdb->users}` AS `u`";

if ( 'last_activity_date' === $this->order_by ) {
	$sql .= "
		LEFT JOIN `{$wpdb->comments}` AS `c`
		ON `u`.`ID` = `c`.`user_id` AND `c`.`comment_type` = 'sensei_course_status'";
}
```

Reply:
I should note that I did test the above with a few pages of users and it does work.

Reply:
My main concern about the latest solution is related to performance.
Have you tested it on large databases (with a lot of students)?
Maybe @Imran92 could advise here as he put a lot of time discovering performance issues in reports.

Reply:
Hey there! I think I had the same issue opened here with some more context: https://github.com/Automattic/sensei/issues/5613

I also added the origin of the problem (handling envs where we can't make relationships between users tables and other tables) and some suggestions on how to fix it.

I'm also adding the issue to this PR to be closed when this is merged.

Reply:
Thanks @renatho !  It's been on the backburner for a while - of the options you listed, which one would you go for?  Just remvoing the sorting, or having a conditional query?

Reply:
@mikeyarce, I'd say that the conditional would be the best approach, so we offer the best possible solution to each scenario, but if we want just to make a quick solution, we could just remove the sorting for now and enhance it in the future.